### PR TITLE
Proposal: HPA Status Conditions

### PR DESCRIPTION
**Pull request by DirectXMan12**
This proposal details an extension to the HPA status field that would
add conditions, similar to other Kubernetes objects.

### CODE REVIEW

**Comment by DirectXMan12** [comment]
cc @kubernetes/sig-autoscaling-feature-requests 
cc @derekwaynecarr @spadgett @jwforres

File: contributors/design-proposals/hpa-status-conditions.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  the HPA object.
+
+New Conditions Types
+--------------------
+
+The above conditions can be coalesced into several condition types. Each
+condition has one or more associated `Reason` values which map back to
+some of the events described above.
+
+- *CanScale*: this condition, when false, indicates issues actually
```
Comment (mwielgus) [inline contributors/design-proposals/hpa-status-conditions.md]: This seams more like "CanAccessScaleRef". "CanScale" is super generic.

File: contributors/design-proposals/hpa-status-conditions.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+- *InBackoff*: this condition, when true, indicates that the HPA is
+  currently within a "scale forbidden window", and therefore will not
+  perform scale operations in a particular direction.  Potential `Reason`
+  values include `BackoffBoth`, `BackoffDownscale`, and `BackoffUpscale`.
+- *CanComputeReplicas*: this condition, when false, indicates issues
+  computing the desired replica counts.  Potential `Reason` values include
+  `FailedGet{Object,Pods,Resource}Metric`, `InvalidMetricSourceType`, and
+  `InvalidSelector` (which includes both missing and unparsable selectors,
+  which can be detailed in the `Message` field).
+
```
Comment (mwielgus) [inline contributors/design-proposals/hpa-status-conditions.md]: So we don't cover situations when HPA would like to go beyond specified min/max?

**Comment by DirectXMan12** [comment]
@kubernetes/sig-autoscaling-feature-requests any further comments here, or is this good to merge?

**Pull request merged**